runner/pkg/metrics/broker.go

```diff
@@ -0,0 +1,92 @@
+// Package metrics provides Prometheus compatible monitoring.
+package metrics
+
+import (
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Broker manages Prometheus metrics.
+type Broker struct {
+	port         int
+	Streams      *prometheus.GaugeVec
+	StreamErrors *prometheus.CounterVec
+}
+
+// Option represents a functional option for configuring a Broker.
+type Option func(broker *Broker)
+
+// NewBroker initializes a new Broker with optional configurations.
+// It sets up Prometheus all available metrics.
+//
+// Example usage:
+//
+//	b := metrics.NewBroker(metrics.WithPort(8080))
+//	go b.Run()
+//
+//	b.Streams.With(b.With().Stream(123).Input("rtmp://1.2.3.4/src")).Set(5)  // Set active streams
+func NewBroker(options ...Option) *Broker {
+	b := &Broker{
+		port: 9947,
+		Streams: promauto.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: "runner",
+			Subsystem: "stream",
+			Name:      "n_streams",
+			Help:      "Number of active streams",
+		}, []string{"stream_id", "input"}),
+
+		StreamErrors: promauto.NewCounterVec(prometheus.CounterOpts{
+			Namespace: "runner",
+			Subsystem: "stream",
+			Name:      "n_errors",
+			Help:      "Number of stream ffmpeg errors",
+		}, []string{"stream_id", "input"}),
+	}
+	for _, option := range options {
+		option(b)
+	}
+	return b
+}
+
+// WithPort configures the Broker to listen on a custom port.
+func WithPort(port int) Option {
+	return func(broker *Broker) {
+		broker.port = port
+	}
+}
+
+// Run starts an HTTP server that exposes the metrics.
+func (b *Broker) Run() {
+	http.Handle("/metrics", promhttp.Handler())
+	slog.Error("Serving metrics", "err", http.ListenAndServe(fmt.Sprintf(":%d", b.port), nil))
```

Comment:
Is this really an `Error`?
Seems a bit sus, but would need an IDE to check further